iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/CalculatedBusImpl.java

```diff
@@ -188,6 +188,23 @@ public Component getSynchronousComponent() {
         return terminalRef == null ? null : scm.getComponent(terminalRef.getSynchronousComponentNumber());
     }
 
+    @Override
+    public VoltageLimits getVoltageLimits() {
+        return getConnectableStream(BusbarSection.class)
+                .map(VoltageLimitsHolder::getVoltageLimits)
+                .filter(Objects::nonNull)
+                .findFirst() // we suppose that all voltage limits in a same calculated bus are equals
+                .orElse(null);
+    }
+
+    @Override
+    public VoltageLimitsAdder newVoltageLimits() {
+        return getConnectableStream(BusbarSection.class)
+                .findFirst()
+                .map(VoltageLimitsHolder::newVoltageLimits)
+                .orElseThrow(() -> new ValidationException(this, "No busbarsection is associated to this calculated bus: no voltage limit can be created."));
+    }
```

Comment:
I would throw an exception in all case: we could probably accept that limits have to be created from "configured" objects (BusbarSections and ConfiguredBus). It's arguable.